### PR TITLE
CSHARP-4193: Add [Beta] flags for public Queryable Encryption API.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: src/MongoDB.Driver/Encryption/AutoEncryptionOptions.cs [hunk 1]
```diff
@@ -109,6 +109,10 @@ public class AutoEncryptionOptions
         /// <value>
         /// The extra options.
         /// </value>
+        /// <remarks>
+        /// All MongoClient objects in the same process should use the same setting for extraOptions.cryptSharedLibPath,
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver/Encryption/AutoEncryptionOptions.cs:113]: I missed that spec required adding this xml comment

File: src/MongoDB.Driver/Encryption/EncryptionAlgorithm.cs [hunk 2]
```diff
@@ -29,11 +29,15 @@ public enum EncryptionAlgorithm
         /// </summary>
         AEAD_AES_256_CBC_HMAC_SHA_512_Random,
         /// <summary>
-        /// Indexed algorithm.
+        /// [Beta] Indexed algorithm.
         /// </summary>
+        /// <remarks>
+        /// To insert or query with an "Indexed" encrypted payload, use a MongoClient configured with AutoEncryptionOptions.
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver/Encryption/EncryptionAlgorithm.cs:35]: I missed that spec required adding this xml comment

**Review by rstam — APPROVED** [review]
LGTM

With the caveat that I have no way of knowing if you missed any places.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**